python/lsst/multiprofit/transforms.py

```diff
@@ -0,0 +1,152 @@
+
+__all__ = ["get_logit_limited", "verify_transform_derivative", "transforms_ref"]
+
+
+from typing import Any
+
+import lsst.gauss2d.fit as g2f
+import numpy as np
+
+from .limits import limits_ref
+
+
+def get_logit_limited(lower: float, upper: float, factor: float = 1.0, name: str | None = None):
+    """Get a logit transform stretched to span a different range than [0,1].
+
+    Parameters
+    ----------
+    lower
+        The lower limit of the range to span.
+    upper
+        The upper limit of the range to span.
+    factor
+        A multiplicative factor to apply to the transformed result.
+    name
+        A descriptive name for the transform.
+
+    Returns
+    -------
+    transform
+        A modified logit transform as specified.
+    """
+    return g2f.LogitLimitedTransformD(
+        limits=g2f.LimitsD(
+            min=lower,
+            max=upper,
+            name=name
+            if name is not None
+            else f"LogitLimitedTransformD(min={lower}, max={upper}, factor={factor})",
+        ),
+        factor=factor,
+    )
+
+
+def verify_transform_derivative(
+    transform: g2f.TransformD,
+    value_transformed: float,
+    derivative: float | None = None,
+    abs_max: float = 1e6,
+    dx_ratios=None,
+    **kwargs: Any,
+):
+    """Verify that the derivative of a transform class is correct.
+
+    Parameters
+    ----------
+    transform
+        The transform to verify.
+    value_transformed
+        The un-transformed value at which to verify the transform.
+    derivative
+        The nominal derivative at value_transformed.
+        Must equal transform.derivative(value_transformed).
+    abs_max
+        The x value to skip verification if np.abs(derivative) > x.
+    dx_ratios
+        Iterable of signed ratios to set dx for finite differencing.
+        dx = value*ratio (untransformed). Only used if dx is None.
+    **kwargs
+        Keyword arguments to pass to np.isclose when comparing derivatives to
+        finite differences.
+
+    Raises
+    ------
+    RuntimeError
+        Raised if the transform derivative doesn't match finite differences
+        within the specified tolerances.
+
+    Notes
+    -----
+    derivative should only be specified if it has previously been computed for
+    the exact value_transformed, to avoid re-computing it unnecessarily.
+
+    Default dx_ratios are [1e-4, 1e-6, 1e-8, 1e-10, 1e-12, 1e-14].
+    Verification will test all ratios until at least one passes.
+    """
+    # Skip testing finite differencing if the derivative is very large
+    # This might happen e.g. near the limits of the transformation
+    # TODO: Check if better finite differencing is possible for large values
+    if abs_max is None:
+        abs_max = 1e8
+    value = transform.reverse(value_transformed)
+    if derivative is None:
+        derivative = transform.derivative(value)
+    is_close = np.abs(derivative) > abs_max
```

Comment:
This is a bit confusing. I would recommend something like
```suggestion
if np.abs(derivative) > abs_max:
    # Skip testing finite differencing if the derivative is very large
    return
is_close = True
```